### PR TITLE
chore: Remove `SidebarNav variant="FULL"`

### LINKED ISSUE #326: SidebarNav: Consider removing the `FULL` variant
https://github.com/gatsby-inc/gatsby-interface/blob/8038a56acf5eda42ebe72c40c048dde080b4695b/src/components/SidebarNav/SidebarNav.tsx#L63

Should we allow setting `maxWidth` instead?
This feels very "magic number", also IMO the variant name is misleading/confusing.

### PR DESCRIPTION
**Pull request by fk**
…and thus all variants of the component:

- Not used in Gatsby Cloud.
- `maxWidth: 8rem` that the `FULL` variant sets on `SidebarNavList` could be set on its parent `SidebarNav` via Emotion's `css` prop or `styled` (at `10rem`, taking `SidebarNav`s `padding-left` into account).
- Remove `Variants` story, knob in `Sandbox` story.

Fixes #326

### CODE REVIEW

**Comment by fk** [comment]
@greglobinski Could you take a look at this? Anything I'm missing?

File: src/components/SidebarNav/SidebarNav.tsx [hunk 1]
```diff
@@ -47,11 +40,9 @@ export default function SidebarNav({
   )
 }
 
-type SidebarNavListProps = JSX.IntrinsicElements["ul"] & {
-  variant?: SidebarNavVariant
-}
+type SidebarNavListProps = JSX.IntrinsicElements["ul"]
 
-function SidebarNavList({ variant = `DEFAULT`, ...rest }: SidebarNavListProps) {
+function SidebarNavList({ ...rest }: SidebarNavListProps) {
```
Comment (greglobinski) [inline src/components/SidebarNav/SidebarNav.tsx:45]: After removing `variant` there is no `rest` and we do not need to destructure the the parameters.

Comment (fk) [reply src/components/SidebarNav/SidebarNav.tsx:45]: 👍 🙏

Comment (fk) [reply src/components/SidebarNav/SidebarNav.tsx]: d158baf

**Pull request merged**